src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
     if (intersection.size() == 2)
         return 1;
 
-    // at least three roads
     std::size_t best = 0;
     double best_deviation = 180;
-
     std::size_t best_continue = 0;
     double best_continue_deviation = 180;
 
-    const EdgeData &in_data = node_based_graph.GetEdgeData(via_edge);
-    const auto in_classification = in_data.road_classification;
+    const EdgeData &in_way = node_based_graph.GetEdgeData(via_edge);
 
-    for (std::size_t i = 1; i < intersection.size(); ++i)
-    {
-        const double deviation = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
-        if (!intersection[i].entry_allowed)
-            continue;
+    std::vector<out_way> out_ways;
 
-        const auto out_data = node_based_graph.GetEdgeData(intersection[i].eid);
-        const auto continue_class =
-            node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification;
+    // create sortable vector of only entry allowed out_ways
+    // get a count of number of ways from that intersection that qualify to have
+    // continue instruction because they share a name with the approaching way
+    std::int64_t continue_count = 0, continue_count_valid = 0;
+    for (std::size_t i = 1, last = intersection.size(); i < last; ++i)
+    {
+        out_way out;
+        const auto out_node = node_based_graph.GetEdgeData(intersection[i].eid);
 
-        const auto same_name = !util::guidance::requiresNameAnnounced(
-            in_data.name_id, out_data.name_id, name_table, street_name_suffix_table);
+        out.road = &intersection[i];
```

Comment:
why do we access the address of the road here? isn't it good enough to store the `index`?
the value here is accessed on two locations. Once to check for entry (which is unnecessary right now, you only keep allowed entries, but see my other two comments on still keeping them) and to check for the edge ID.

I'd rather store the two values directly (even less memory required), and avoid keeping these pointers.